### PR TITLE
Update the workflow sections

### PR DESCRIPTION
**Pull request by adbridge**
This commit updates the pull request sections to detail the new
pull request template.

### CODE REVIEW

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -61,90 +61,142 @@ If commits do not follow the above guidelines, we may request that you modify th
...
-Release: patch
+#### Feature update
+
+This can be any change in the functionality, including adding a new feature, a new method or a function. Software language does not matter.
+
+A feature contribution contains a new API, capability or behavior. It does not break backward compatibility with existing APIs, capabilities or behaviors. New feature contributions are very welcome in Mbed OS. However, because they add capability to the codebase, a new feature may introduce bugs and a support burden. New features should also come with documentation, support for most targets and comprehensive test coverage. Feature PRs are treated cautiously, and new features require a new minor version for the codebase. Features are candidates for feature releases.
+
+We initially implement new features on separate branches in the Mbed OS repository. Mbed OS maintainers or 
+tech leads may create the new branches by following the naming convention: "feature-" prefix.
+Each
```
Comment (0xc0170) [inline docs/reference/contributing/guidelines/workflow.md]: Each 2x?

File: docs/reference/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -61,90 +61,142 @@ If commits do not follow the above guidelines, we may request that you modify th
...
+##### Test update
 
 Test updates include updates to a new test unit or test case.
 
-Release: patch
+#### Feature update
+
+This can be any change in the functionality, including adding a new feature, a new method or a function. Software language does not matter.
+
+A feature contribution contains a new API, capability or behavior. It does not break backward compatibility with existing APIs, capabilities or behaviors. New feature contributions are very welcome in Mbed OS. However, because they add capability to the codebase, a new feature may introduce bugs and a support burden. New features should also come with documentation, support for most targets and comprehensive test coverage. Feature PRs are treated cautiously, and new features require a new minor version for the codebase. Features are candidates for feature releases.
```
Comment (0xc0170) [inline docs/reference/contributing/guidelines/workflow.md]: should this contain also this: "The Mbed OS technical lead must approve feature change pull requests.", this is currently what we do

**Comment by 0xc0170** [comment]
@AnotherButler is there chance to get this in by end of this week ? No dependency here, the template was integrated in Mbed OS already to start using simplified PR type.

File: docs/reference/contributing/guidelines/workflow.md (hunk 1 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Well spotted!

File: docs/reference/contributing/guidelines/workflow.md (hunk 2 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Yes we can add that.

File: docs/reference/contributing/guidelines/workflow.md [hunk 3]
```diff
@@ -61,90 +61,143 @@ If commits do not follow the above guidelines, we may request that you modify th
...
+A bug fix is a change that fixes a specific defect in the codebase with backward compatibility. These are the highest priority because of the positive effect the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Every bug fix should contain a test to verify results before and after the pull request. Bug fixes are candidates for patch releases. Large, nontrivial bug fixes approaching the size of refactors run the risk of being considered refactors themselves.
+
+##### Refactor
 
-Release: patch
+A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this are moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes and test organization changes. There is always the risk that someone depended on the location or name before a refactor; therefore, these are lower in priority than bug fixes and might require detailed justification for the change. Refactors are candidates for feature releases.
```
Comment (0xc0170) [inline docs/reference/contributing/guidelines/workflow.md]: > Refactors are candidates for feature releases.

this one should be updated to reflect PR type above - patch release

**Review by 0xc0170 — CHANGES_REQUESTED** [review]
One change request, otherwise LGTM

File: docs/reference/contributing/guidelines/workflow.md (hunk 3 above)
Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md]: Ah just need to remove that last sentence completely as it is already under the Patch section

**Review by 0xc0170 — APPROVED** [review]

**Comment by AnotherButler** [comment]
@0xc0170 After I've finished the priority, planned work, I'll see what I can do.

**Comment by 0xc0170** [comment]
Any update for this one?

File: docs/reference/contributing/guidelines/workflow.md [hunk 4]
```diff
@@ -61,90 +61,131 @@ If commits do not follow the above guidelines, we may request that you modify th
 
 [How We Release Arm Mbed OS](../introduction/how-we-release-arm-mbed-os.html) explains Mbed OS versioning.
 
-## Pull request types
+## Pull request template
```
Comment (AnotherButler) [inline docs/reference/contributing/guidelines/workflow.md:64]: The blog post says the template now "includes a note about reporting and questions and enhancements to the Forum," but I don't see that here.

Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md:64]: Ah I think the blog post might have been a little confusing. When someone raises an issue that is not a bug then the git2jira bot closes the issue with a comment stating that questions or enhancements should be raised in the forums. It has nothing to do with PRs

Comment (adbridge) [reply docs/reference/contributing/guidelines/workflow.md:64]: In the issue template:
(which is not affected by this PR)
************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

Comment (AnotherButler) [reply docs/reference/contributing/guidelines/workflow.md:64]: Thanks for clarifying 👍 
Should we mention somewhere (not in the PR section) that nonbug issues will be closed?

Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md:64]: > You can submit Mbed OS bugs on the forums or directly on GitHub.

I'll send separate PR to change this in this workflow file

**Review by 0xc0170 — APPROVED** [review]

File: docs/reference/contributing/guidelines/workflow.md (hunk 4 above)
Comment (0xc0170) [reply docs/reference/contributing/guidelines/workflow.md:64]: Here https://github.com/ARMmbed/mbed-os-5-docs/pull/1161

**Review by AnotherButler — APPROVED** [review]
Thanks for the PRs, y'all 👍

**Comment by AnotherButler** [comment]
Could you please fix the merge conflicts, so I can merge?

**Comment by adbridge** [comment]
@AnotherButler conflicts sorted :)

**Comment by AnotherButler** [comment]
Thanks, @adbridge 👍

**Pull request merged**